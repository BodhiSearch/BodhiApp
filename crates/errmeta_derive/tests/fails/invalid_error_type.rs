pub struct InvalidErrorType;

#[derive(Debug, errmeta_derive::ErrorMeta)]
enum TestError {
  #[error_meta(code = "bad_request", status = 200, error_type = InvalidErrorType)]
  BadRequest,
}

fn main() {}
