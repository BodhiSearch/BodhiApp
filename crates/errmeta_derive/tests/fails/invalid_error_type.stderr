error[E0277]: the trait bound `InvalidErrorType: AsRef<str>` is not satisfied
 --> tests/fails/invalid_error_type.rs:3:17
  |
3 | #[derive(Debug, errmeta_derive::ErrorMeta)]
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `AsRef<str>` is not implemented for `InvalidErrorType`
  |
  = note: this error originates in the derive macro `errmeta_derive::ErrorMeta` (in Nightly builds, run with -Z macro-backtrace for more info)
